sdk/src/main/java/com/google/cloud/dataflow/sdk/io/range/OffsetRangeTracker.java

```diff
@@ -76,8 +88,14 @@ public synchronized boolean tryReturnRecordAt(boolean isAtSplitPoint, long recor
               recordStart,
               lastRecordStart));
     }
+
+    if (isAtSplitPoint && lastRecordStart >= -1) {
+      ++seenSplitPoints;
```

Comment:
Yes, there are no new tests yet ;) -- this was intended for high-level feedback.